src/wallet/wallet.cpp

```diff
@@ -2346,6 +2346,7 @@ std::map<CTxDestination, std::vector<COutput>> CWallet::ListCoins() const
 
 const CTxOut& CWallet::FindNonChangeParentOutput(const CTransaction& tx, int output) const
 {
+    AssertLockHeld(cs_wallet);
```

Comment:
@achow101 per https://github.com/bitcoin/bitcoin/pull/19738#issuecomment-675602253 I'm curious why this should be added, as there is already an `AssertLockHeld(cs_wallet);` at the top of its caller, `ListCoins()`. For a future call from elsewhere?

Reply:
This was based on #19289 which then raised #19773, and @achow101 suggestion should be done there, `ListCoins` eventually calls `FindNonChangeParentOutput`, and like `IsTrusted`, was missing the assertion.

Reply:
@jonatack I could split to a different commit though to be more correct, just LMK.

Reply:
No worries, I was only trying to understand the locking. This is fine.